doc/tutorials/contributing-clion.md

```diff
@@ -0,0 +1,183 @@
+# Introduction
+
+In this tutorial, we will go through the steps necessary to contribute to
+Elektra to make it easier for you to get started. We will use a Unix based OS
+like Linux and CLion for development, but depending on where you want to
+contribute code, other IDEs will also be sufficient. Before you start, please
+read [this](/.github/CONTRIBUTING.md) to get familiar with the process of
+contributing to Elektra.
+
+## Prerequisites
+
+- Git ([link](https://git-scm.com/download)) or similar software
+- CLion ([link](https://www.jetbrains.com/clion/)) or similar software
+- GitHub-account ([link](https://github.com/))
+
+## Forking the repository
+
+Libelektra is hosted on GitHub. You can find its repository here:
+
+- https://github.com/ElektraInitiative/libelektra
+
+To be able to make pull requests, you need a copy of this repository inside
+your GitHub-account. You can find a tutorial about how to do this
+[here](https://help.github.com/en/articles/fork-a-repo) (remember to sign
+into your account first). After this, you should see this copy in the list
+of your own repositories with a hint of it's origin.
+
+## Getting the source-code
+
+To develop for libelektra, we now have to "download" your copy of it's
+original repository. In git this process is called "cloning". Open a
+terminal and navigate to the folder you want to save the source code into
+and type:
+
+```sh
+git clone https://github.com/ElektraInitiative/libelektra.git
+```
+
+With the project now locally available we can start developing.
+
+## Setting up the project
+
+It is important to start CLion as root-user, since some functionalities of
```

Comment:
I actually wouldn't recommend that...

Instead I would recommend changing the CMake configurations such that you don't need root access.

To do that open `File | Settings | Build, Execution, Deployment | CMake`. There you should see all the different CMake configurations created (and automatically refreshed) by CLion.  For most configurations, the CMake options field  should contain (click the icon on the right to see one option per line) the following to remove need for root access:

```
-DKDB_DB_SYSTEM="~/.config/kdb/[xyz]/system"
-DKDB_DB_SPEC="~/.config/kdb/[xyz]/spec"
-DKDB_DB_USER=".config/kdb/[xyz]/user"
-DCMAKE_INSTALL_PREFIX=install
-DINSTALL_SYSTEM_FILES=OFF
``` 
Replace `[xyz]` by some unique name for the configuration e.g. dbg for debug. We use this, so that the different configurations don't clash with each other. The `CMAKE_INSTALL_PREFIX` changes the destination of `make install` to the `install` folder inside the CMake build directory. The advantage of using these lines, is that the resulting build of Elektra is completely isolated and shouldn't interfere with existing Elektra installations (even if `make install` is called).

Since this changes some of Elektra's behaviour (e.g. spec and system namespace are now also user specific), I don't recommend using these options for release builds. If you want to build Elektra, for use outside of development (of Elektra), I recommend just building via command line and sudo.

I would also recommend adding the options `-DENABLE_DEBUG=ON` and `-DENABLE_LOGGER=ON` for any debug builds, to enable some additional checks and logging capabilities. Additional, you may improve build speeds by setting the build options field to e.g. `-j 8` (or more depending on your processor).

Users/developers can of course also add their own options.

Another note: While you may add as many different configurations as you want, Clion automatically maintains them all. This means everytime you change CMake files Clion will run `cmake` for all these configurations **in parallel**. Too many configurations and your system will grind to a halt for a few seconds everytime Clion decides to reload the CMake project.